platform/security/idp/security-idp-server/src/main/java/org/codice/ddf/security/idp/server/IdpEndpoint.java

```diff
@@ -0,0 +1,820 @@
+    private static final String HTTP_ARTIFACT_BINDING = "urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Artifact";
+
+    private PKIAuthenticationTokenFactory tokenFactory;
+
+    private SecurityManager securityManager;
+
+    private Crypto signatureCrypto;
+
+    private Crypto encryptionCrypto;
+
+    private EncryptionService encryptionService;
+
+    private String signaturePropertiesPath;
+
+    private String encryptionPropertiesPath;
+
+    private Map<String, EntityDescriptor> serviceProviders = new HashMap<>();
+
+    private String indexHtml;
+
+    private String submitForm;
+
+    private String redirectPage;
+
+    private Boolean strictSignature = true;
+
+    public void init() {
+        try {
+            indexHtml = IOUtils.toString(IdpEndpoint.class.getResourceAsStream("/html/index.html"));
+            submitForm = IOUtils.toString(
+                    IdpEndpoint.class.getResourceAsStream("/templates/submitForm.handlebars"));
+            redirectPage = IOUtils.toString(
+                    IdpEndpoint.class.getResourceAsStream("/templates/redirect.handlebars"));
+        } catch (Exception e) {
+            LOGGER.error("Unable to load index page for IDP.", e);
+        }
+    }
+
+    private void parseSpMetadata(List<String> spMetadata) {
+        if (spMetadata != null) {
+            try {
+                serviceProviders = MetadataConfigurationParser.buildEntityDescriptors(spMetadata);
+            } catch (IOException e) {
+                LOGGER.error("Unable to parse SP metadata configuration.", e);
+            }
+        }
+    }
+
+    @POST
+    public Response showPostLogin(@FormParam(SAML_REQ) String samlRequest,
+            @FormParam(RELAY_STATE) String relayState, @Context HttpServletRequest request,
+            @Context HttpServletResponse response) throws WSSecurityException {
+        LOGGER.debug("Recevied POST IdP request.");
+        return showLoginPage(samlRequest, relayState, null, null, request, true);
+    }
+
+    @GET
+    public Response showGetLogin(@QueryParam(SAML_REQ) String samlRequest,
+            @Encoded @QueryParam(RELAY_STATE) String relayState,
+            @QueryParam(SSOConstants.SIG_ALG) String signatureAlgorithm,
+            @QueryParam(SSOConstants.SIGNATURE) String signature,
+            @Context HttpServletRequest request, @Context HttpServletResponse response)
+            throws WSSecurityException {
+        LOGGER.debug("Recevied GET IdP request.");
+        return showLoginPage(samlRequest, relayState, signatureAlgorithm, signature, request, false);
+    }
+
+    private Response showLoginPage(String samlRequest, String relayState, String signatureAlgorithm,
+            String signature, HttpServletRequest request, boolean isPost)
+            throws WSSecurityException {
+        String responseStr;
+        AuthnRequest authnRequest = null;
+        try {
+            Map<String, Object> responseMap = new HashMap<>();
+            validateRelayState(relayState);
```

Comment:
Can we extract a separate class to deal with all the parameter validation and auth related stuff? This would shrink this method and class some, keep all that logic together and help with unit testing.